jsx/Form.js

```diff
@@ -132,20 +109,53 @@ var FormElement = React.createClass({
       </form>
     );
   }
-});
+}
+
+FormElement.propTypes = {
+  name: PropTypes.string.isRequired,
+  id: PropTypes.string,
+  method: PropTypes.oneOf(['POST', 'GET']),
+  action: PropTypes.string,
+  class: PropTypes.string,
+  columns: PropTypes.number,
+  formElements: PropTypes.shape({
+    elementName: PropTypes.shape({
+      name: PropTypes.string,
+      type: PropTypes.string
+    })
+  }),
+  onSubmit: PropTypes.func,
+  onUserInput: PropTypes.func
+}
+
+FormElement.defaultProps = {
+  name: null,
+  id: '',
+  method: 'POST',
+  action: undefined,
+  class: 'form-horizontal',
+  columns: 1,
+  fileUpload: false,
+  formElements: {},
+  onSubmit: function() {
+    console.warn('onSubmit() callback is not set!');
+  }
+};
+
 
 /**
  * Search Component
  * React wrapper for a searchable dropdown
  */
-class SearchableDropdown extends React.Component {
-  constructor(props) {
-    super(props);
+class SearchableDropdown extends Component {
```

Comment:
extends React.Component should remain the same. similarly to props within constructor() and super()